lib/file_store/s3_store.rb

```diff
@@ -19,12 +19,16 @@ def initialize(s3_helper = nil)
 
     def store_upload(file, upload, content_type = nil)
       path = get_path_for_upload(upload)
-      store_file(file, path, filename: upload.original_filename, content_type: content_type, cache_locally: true)
+      url, etag = store_file(file, path, filename: upload.original_filename, content_type: content_type, cache_locally: true)
+      upload.update_attributes!(etag: etag)
```

Comment:
Do we need to save here? I think the upload record is expected to be saved by the client as it needs to save the url too.

Reply:
[here](https://github.com/discourse/discourse/blob/86255faa08ed01930b55888cf52bc75d339f9766/lib/upload_creator.rb#L129) we only updating the url attribute. In this case I'll to update that too.

Reply:
Yea just make the method do a `save!`

Reply:
Done.